py/selenium/webdriver/remote/webdriver.py

```diff
@@ -842,15 +844,20 @@ def get_screenshot_as_file(self, filename):
             return False
         finally:
             del png
+        if not filename.lower().endswith('.png'):
```

Comment:
Can you move this warning to be the first thing that happens in the method so people can see the warning _and_ get any exceptions that might happen.